app/src/components/Exercise/CreateExercisePopUp.tsx

```diff
@@ -1,199 +1,137 @@
-// Pop-up messages
-import react from 'react';
-import { toast } from 'react-toastify';
-import { useState } from 'react'
-import { Navigate, useNavigate } from 'react-router-dom';
-import { useForm, type SubmitHandler } from 'react-hook-form'
+import { useState, } from "react";
+import { useForm, SubmitHandler } from "react-hook-form";
 
 // Icons
-import { PlusIcon } from "@heroicons/react/24/outline";
 import { PencilSquareIcon } from '@heroicons/react/24/outline'
 
+
 // Components
-import { AnswerField } from "../../components/Exercise/AnswerField";
-import { Answer } from "../../interfaces/AnswerClass";
-import { Exercise } from "../../interfaces/Exercise";
+import AnswerCards from "../../components/Exercise/AnswerCards";
+
+// Interfaces
+import { Answer } from "../../interfaces/Answer";
+import { Exercise } from "../../interfaces/Exercise"
 
-// Services
-import ExerciseServices from '../../services/exercise.services';
+// Helpers
+import ExerciseServices from "../../services/exercise.services";
 
+// Pop-up messages
+import { toast } from "react-toastify";
 
+// Hooks
+import useToken from "../../hooks/useToken";
 
-type ExercisePartial = {
-    title: string;
-    question: string;
-    answers: Answer[];
-};
+
+export interface ExercisePartial {
+    title: string,
+    question: string,
+    answers: Answer[]
+}
 
 type Inputs = {
     sid: string|undefined;
-    cid: string|undefined;
-  
 };
 
 
+export const CreateExercise = ({sid}:Inputs) => {
 
-export const CreateExercise = ({sid, cid}:Inputs) => {
-    const [isLoading, setIsLoading] = useState(false);
-    const [tempAnswers, setTempAnswer] = useState<Answer[]>([new Answer(), new Answer()]);
+    let TempAnswers = [{text: "", isCorrect: true, feedback: ""}, {text: "", isCorrect: false, feedback: ""}];
 
+    const [answers, setAnswers] = useState<Answer[]>(TempAnswers);
 
-    const [answerFieldIndex, setAnswerFieldIndex] = useState(tempAnswers.length);
-    const [answerField, setAnswerField] = useState<JSX.Element[]>([<AnswerField index={0} answerObject={tempAnswers[0]}  callback={updateAnswer}/>, <AnswerField index={1} answerObject={tempAnswers[1]}  callback={updateAnswer}/>]);
-    const navigate = useNavigate();
+    const { register, handleSubmit: handleExerciseSave, formState: { errors } } = useForm();
+    const onExerciseSave: SubmitHandler<any> = data => createExercise(data);
+
+    /** Token doesnt work, reimplement when it token is implemented */
+    //const token = useAuthStore(state => state.token);
+    //const token = useToken();
     const token = "dummyToken";
```

Comment:
For now, to get the token you can write `const token = localStorage.getItem("token")`
Don't know how important it is for your PBIs, but still.

Reply:
Hey have checked it out, our code is not set up to take a null value, which the "localStorage.getItem("token")" can return so for now we will not use it.